python-package/geobr/utils.py

```diff
@@ -218,4 +232,33 @@ def get_metadata(geo, data_type, year):
     # Verify year input
     metadata = apply_year(metadata, year)
 
-    return metadata
\ No newline at end of file
+    return metadata
+
+
+def list_geobr_functions():
```

Comment:
Can you add a test to it?

Reply:
I'm not sure how to do it.

Reply:
In the `tests` folder, you are going to find a file called `test_utils.py`. There, you'll find all the current tests. Basically, the idea here is to test if the function is behaving as expected. Since this function just prints stuff, I'd do the following tests:

- check if it is printing something https://docs.pytest.org/en/latest/capture.html
- check if the url that we are fetching data exists i.e. gives code `200`.

To run the tests, just run `pytest` in the python root folder

Reply:
João, Thanks. Everybody recommends using capsys. However, that is a parameter that you set when you run pytest from the terminal. How do you access capsys so that you can call a function with it: `def test_function(capsys)` if you can't import `capsys`. Sorry, if it sounds obvious.

Reply:
I have never used it. But, from the examples, I don't think you have to import it.  It comes with pytest itself https://docs.pytest.org/en/latest/reference.html#_pytest.capture.capsys

Reply:
I wasn't able to test it. I don't know how to invoke it. Please, see the uploaded version and this question [I posted at StackOverflow](https://stackoverflow.com/questions/60345014/accessing-pytest-capsys-from-inside-a-test?noredirect=1#comment106759467_60345014)

Reply:
However, when I ran `python --m pytest ` there were no errors! See output below.

```====================================================================== test session starts ======================================================================
platform win32 -- Python 3.7.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: C:\Users\r1702898\Documents\Modelagem\geobr\python-package
collected 9 items                                                                                                                                                

tests\test_read_state.py .                                                                                                                                 [ 11%]
tests\test_utils.py ........                                                                                                                               [100%]

======================================================================= 9 passed in 5.64s =======================================================================`` `

Reply:
All double-checked. Function is printing and checking url exists! Thanks, just learned how to use pytest!